Assets/MRTK/SDK/Features/UX/Scripts/RiggedHandVisualizer/RiggedHandVisualizer.cs

```diff
@@ -301,178 +287,143 @@ private void OnDisable()
-            if (handTrackingProfile != null && handTrackingProfile.EnableHandJointVisualization)
-            {
-                // This starts at 1 to skip over TrackedHandJoint.None.
-                for (int i = 1; i < ArticulatedHandPose.JointCount; i++)
+                // Ensures that the hand only renders when the event data matches the controller this visualizer represents
+                if (eventData.InputSource.SourceId != Controller.InputSource.SourceId)
                 {
-                    TrackedHandJoint handJoint = (TrackedHandJoint)i;
-                    MixedRealityPose handJointPose = eventData.InputData[handJoint];
-
-                    if (skeletonJoints.TryGetValue(handJoint, out Transform skeletonJointTransform))
-                    {
-                        skeletonJointTransform.SetPositionAndRotation(handJointPose.Position, handJointPose.Rotation);
-                    }
-                    else
-                    {
-                        GameObject prefab;
-                        if (handJoint == TrackedHandJoint.Palm)
-                        {
-                            prefab = inputSystem.InputSystemProfile.HandTrackingProfile.PalmJointPrefab;
-                        }
-                        else if (handJoint == TrackedHandJoint.IndexTip)
-                        {
-                            prefab = inputSystem.InputSystemProfile.HandTrackingProfile.FingerTipPrefab;
-                        }
-                        else
-                        {
-                            prefab = inputSystem.InputSystemProfile.HandTrackingProfile.JointPrefab;
-                        }
-
-                        GameObject jointObject;
-                        if (prefab != null)
-                        {
-                            jointObject = Instantiate(prefab);
-                        }
-                        else
-                        {
-                            jointObject = new GameObject();
-                        }
+                    return;
+                }
+                Debug.Assert(eventData.Handedness == Controller.ControllerHandedness);
 
-                        jointObject.name = handJoint.ToString() + " Proxy Transform";
-                        jointObject.transform.SetPositionAndRotation(handJointPose.Position, handJointPose.Rotation);
-                        jointObject.transform.parent = transform;
+                IMixedRealityInputSystem inputSystem = CoreServices.InputSystem;
+                MixedRealityHandTrackingProfile handTrackingProfile = inputSystem?.InputSystemProfile.HandTrackingProfile;
 
-                        skeletonJoints.Add(handJoint, jointObject.transform);
-                    }
-                }
-            }
-            else
-            {
-                // clear existing joint GameObjects / meshes
-                foreach (var joint in skeletonJoints)
+                if (receivingPlatformHandMesh)
```

Comment:
Should/could this be even higher in the method? Maybe even up on line 304 `if (receivingPlatformHandMesh || eventData.InputSource.SourceId != Controller.InputSource.SourceId)`?

Otherwise, it looks like we're doing a couple things (like grabbing the `InputSystem`) just to return early after

Reply:
Done!